src/main/java/Sinep/Deadline.java

```diff
@@ -0,0 +1,19 @@
+package Sinep;
+public class Deadline extends Task {
+    protected String dateTime;
+    public Deadline(String description) {
+        super(description);
+        String[] givenDeadline = description.split("/by ", 2);
+        //this.description = description;
+        this.description = givenDeadline.length >= 1 ? givenDeadline[0].trim() : "";
+        this.dateTime = givenDeadline.length > 1 ? givenDeadline[1].trim() : "";
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + getStatusIcon() + " " + description + " (by: " + dateTime + ")";
+    }
+
+
+
```

Comment:
You may remove these extra lines. In IntelliJ, the code formatter can be updated to do this for you, every time you hit `CTRL/CMD + S`.